Can not be used, pg revision changed the structure of several files

Can not be used, pg revision changed the structure of several files

C:\Users\Dev1>go get github.com/dizzyfool/genna
# github.com/dizzyfool/genna/database
E:\Work\GO\package\src\github.com\dizzyfool\genna\database\database.go:24:7: event.Data undefined (type *pg.QueryEvent has no field or method Data)
E:\Work\GO\package\src\github.com\dizzyfool\genna\database\database.go:35:10: event.Data undefined (type *pg.QueryEvent has no field or method Data)
E:\Work\GO\package\src\github.com\dizzyfool\genna\database\database.go:36:20: event.Data undefined (type *pg.QueryEvent has no field or method Data)
E:\Work\GO\package\src\github.com\dizzyfool\genna\database\database.go:54:36: cannot use NewQueryLogger(logger) (type QueryLogger) as type pg.QueryHook in argument to client.baseDB.AddQueryHook:
        QueryLogger does not implement pg.QueryHook (wrong type for AfterQuery method)
                have AfterQuery(*pg.QueryEvent)
                want AfterQuery(context.Context, *pg.QueryEvent) (context.Context, error)

--- a/database/database.go
+++ b/database/database.go
@@ -1,6 +1,7 @@
 package database
 
 import (
+	"context"
 	"time"
 
 	"github.com/go-pg/pg"
@@ -20,20 +21,21 @@ func NewQueryLogger(logger *zap.Logger) QueryLogger {
 }
 
 // BeforeQuery stores start time in custom data array
-func (ql QueryLogger) BeforeQuery(event *pg.QueryEvent) {
-	event.Data["startedAt"] = time.Now()
+func (ql QueryLogger) BeforeQuery(ctx context.Context, event *pg.QueryEvent) (context.Context, error) {
+	event.Stash["startedAt"] = time.Now()
+	return ctx, nil
 }
 
 // AfterQuery calculates execution time and print it with formatted query
-func (ql QueryLogger) AfterQuery(event *pg.QueryEvent) {
+func (ql QueryLogger) AfterQuery(ctx context.Context, event *pg.QueryEvent) (context.Context, error) {
 	query, err := event.FormattedQuery()
 	if err != nil {
 		ql.logger.Error("formatted query error", zap.Error(err))
 	}
 
 	var since time.Duration
-	if event.Data != nil {
-		if v, ok := event.Data["startedAt"]; ok {
+	if event.Stash != nil {
+		if v, ok := event.Stash["startedAt"]; ok {
 			if startAt, ok := v.(time.Time); ok {
 				since = time.Since(startAt)
 			}
@@ -41,6 +43,7 @@ func (ql QueryLogger) AfterQuery(event *pg.QueryEvent) {
 	}
 
 	ql.logger.Debug(query, zap.Duration("duration", since))
+	return ctx, nil
 }
 
 // NewDatabase creates database connection

